docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,24 @@
+# Charm architecture
+
+The SMTP Integrator charm fetches centralizes SMTP configuration and propagates it through a Juju integration.
+
+## Juju events
+
+According to the [Juju SDK](https://juju.is/docs/sdk/event): "an event is a data structure that encapsulates part of the execution context of a charm".
+
+For this charm, the following events are observed:
+
+1. [config-changed](https://juju.is/docs/sdk/config-changed-event): usually fired in response to a configuration change using the GUI or CLI. Action: validate the configuration and propagate the SMTP configuration through the relation.
+2. [update-status](https://juju.is/docs/sdk/update-status-event): fired periodically. Action: propagate the SMTP configuration through the relation.
+3. [smtp-relation-joined](https://juju.is/docs/sdk/relation-name-relation-joined-event): Custom event for when a new SMTP relations joins. Action: write the SMTP details in the relation databag. The `saml` integration will share a secret id across the relation the requirer will be able to access to retrieve the password.
+4. [smtp-legacy-relation-joined](https://juju.is/docs/sdk/relation-name-relation-joined-event): Custom event for when a new legacy SMTP relations joins. Action: write the SMTP details in the relation databag. The `saml-legacy` integration will share the password across the relation.
+
```

Comment:
```suggestion
3. [smtp-relation-joined](https://juju.is/docs/sdk/relation-name-relation-joined-event): Custom event for when a new SMTP relation joins. Action: write the SMTP details in the relation databag. The `smtp` relation will share a secret id across the relation the requirer will be able to access to retrieve the password.
4. [smtp-legacy-relation-joined](https://juju.is/docs/sdk/relation-name-relation-joined-event): Custom event for when a new legacy SMTP relations joins. Action: write the SMTP details in the relation databag. The `smtp-legacy` relation will share the password across the relation.

```
I am not sure whether the term relation should be replaced by integration in the text, but I have replaced the term integration by relation once for consistency.

Reply:
`integration` is the juju 3 terminology, so it's preferred over `relation`